_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -3430,7 +3414,26 @@ mfxStatus ImplementationAvc::UpdateBitstream(
 
     // Update bitstream fields
     task.m_bs->TimeStamp = task.m_timeStamp;
+    if (fid == 0)
```

Comment:
I believe the best way will be to do calculation of task.m_decodeTimeStamp inside function ConfigureTask.
1) It will keep correct order of picking DTS from m_decodeTimeStamps because ConfigureTask is called for reordered frame (before out-of-order submission optimization takes place).
2) "if (fid == 0)" will not be required there.
3) It will automatically take care about "FieldOutput" mode.